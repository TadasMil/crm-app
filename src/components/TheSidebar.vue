<script setup name="TheSidebar" lang="ts">
  import BaseIcon from './BaseIcon.vue'
  interface IPageItem {
    name: string
    label: string
    icon: string
  }

  const pages: IPageItem[] = [
    {
      name: 'UsersIndex',
      label: 'Users',
      icon: 'users',
    },
  ]
</script>

<template>
  <div class="flex flex-col overflow-hidden bg-primary">
    <div class="flex h-20 items-center justify-center shadow-md shadow-white">
      <h1 class="text-3xl font-bold uppercase tracking-widest text-white">
        CRM
      </h1>
    </div>
    <ul v-for="page in pages" :key="page.name" class="flex flex-col py-4">
      <li
        class="flex items-center space-x-4 px-6 transition-transform duration-200 ease-in hover:translate-x-2 hover:text-secondary"
      >
        <BaseIcon icon="users" class="h-6 w-6 text-white" />
        <router-link
          class="flex h-12 transform flex-row items-center text-lg font-medium tracking-widest text-white outline-none"
          :to="{ name: page.name }"
        >
          {{ page.label }}
        </router-link>
      </li>
    </ul>
  </div>
</template>
