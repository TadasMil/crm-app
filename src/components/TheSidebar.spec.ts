import { render } from '@testing-library/vue'
import { describe, expect, it, vi } from 'vitest'

import TheSidebar from './TheSidebar.vue'

describe('TheSidebar.spec.js', () => {
  it('should render the component', () => {
    const { html } = render(TheSidebar, {
      global: {
        stubs: ['BaseIcon', 'router-link'],
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
