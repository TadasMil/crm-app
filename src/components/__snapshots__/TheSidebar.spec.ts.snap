// Vitest Snapshot v1

exports[`TheSidebar.spec.js > should render the component 1`] = `
"<div class=\\"flex flex-col overflow-hidden bg-primary\\">
  <div class=\\"flex h-20 items-center justify-center shadow-md shadow-white\\">
    <h1 class=\\"text-3xl font-bold uppercase tracking-widest text-white\\"> CRM </h1>
  </div>
  <ul class=\\"flex flex-col py-4\\">
    <li class=\\"flex items-center space-x-4 px-6 transition-transform duration-200 ease-in hover:translate-x-2 hover:text-secondary\\">
      <base-icon-stub icon=\\"users\\" class=\\"h-6 w-6 text-white\\"></base-icon-stub>
      <router-link-stub class=\\"flex h-12 transform flex-row items-center text-lg font-medium tracking-widest text-white outline-none\\" to=\\"[object Object]\\"></router-link-stub>
    </li>
  </ul>
</div>"
`;
